esphome/components/mpl115a2/mpl115a2.cpp

```diff
@@ -0,0 +1,85 @@
+#include "mpl115a2.h"
+#include "esphome/core/hal.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace mpl115a2 {
+
+static const char *const TAG = "mpl115a2";
+
+void MPL115A2Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MPL115A2...");
+
+  this->read_coefficients_();
+}
+
+void MPL115A2Component::read_coefficients_() {
+  int16_t a0coeff;
+  int16_t b1coeff;
+  int16_t b2coeff;
+  int16_t c12coeff;
+
+  uint8_t cmd;
+  uint8_t buffer[8];
+
+  cmd = MPL115A2_REGISTER_A0_COEFF_MSB;
+  this->write(&cmd, 1);
+  this->read(buffer, 8);
+
+  a0coeff = (((uint16_t) buffer[0] << 8) | buffer[1]);
+  b1coeff = (((uint16_t) buffer[2] << 8) | buffer[3]);
+  b2coeff = (((uint16_t) buffer[4] << 8) | buffer[5]);
+  c12coeff = (((uint16_t) buffer[6] << 8) | buffer[7]) >> 2;
```

Comment:
```suggestion
  a0coeff = encode_uint16(buffer[0], buffer[1]);
  b1coeff = encode_uint16(buffer[2], buffer[3]);
  b2coeff = encode_uint16(buffer[4], buffer[5]);
  c12coeff = encode_uint16(buffer[6], buffer[7]) >> 2;
```

This will also require `#include "esphome/core/helpers.h"`